### PR TITLE
Add linting for public documentation

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This PR modifies the detekt configuration to check undocumented public functions and classes. All instances that were offending have been fixed as well with the help of Chad.

# Description
See above. 

# How Has This Been Tested?
No functional changes.

### CODE REVIEW

File: common/src/main/kotlin/Base58.kt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/**
+ * A utility object for Base58 encoding and decoding.
+ */
 public object Base58Btc {
   private val ALPHABET = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
```
Comment (nitro-neal) [inline common/src/main/kotlin/Base58.kt:5]: What is this for?

Comment (nitro-neal) [inline common/src/main/kotlin/Base58.kt:5]: What is this for?

File: credentials/src/main/kotlin/SSI.kt [hunk 2]
```diff
@@ -155,120 +163,150 @@ public data class DecodedVcJwt(
...
-     * @return A [VpJwt] representing the verifiable presentation in JWT format.
-     * @throws Exception if there are no usable verifiable credentials that correspond to the presentation definition or any other exception occurring during the creation of the presentation.
-     */
-    @Throws(Exception::class)
-    public fun create(signOptions: SignOptions, createVpOptions: CreateVpOptions): VpJwt {
+  /**
+   * Verifies the authenticity of a verifiable credential using its JWT representation and a DID resolver.
+   *
+   * @param vcJWT The JWT representation of the verifiable credential to be verified.
+   * @param resolver The DID resolver used for resolving issuer DID information.
```
Comment (nitro-neal) [inline credentials/src/main/kotlin/SSI.kt:228]:
```suggestion
   * @param resolver The DID resolver used for resolving DID information.
```

**Review by nitro-neal — APPROVED** [review]
🆗 ✅

File: common/src/main/kotlin/Base58.kt (hunk 1 above)
Comment (andresuribe87) [reply common/src/main/kotlin/Base58.kt:5]: No idea lol. It was already in there.

Comment (mistermoe) [reply common/src/main/kotlin/Base58.kt:5]: @nitro-neal, there's more context on the crypto branch [here](https://github.com/TBD54566975/web5-sdk-kotlin/blob/crypto/common/src/main/kotlin/web5/sdk/common/Base58Btc.kt)

Currently, Base58Btc is used for `did:key`

**Pull request merged**